botocore/awsrequest.py

```diff
@@ -450,6 +471,92 @@ def prepare_body(self, data, files, json=None):
                 if 'Transfer-Encoding' in self.headers:
                     self.headers.pop('Transfer-Encoding')
 
+        if self.body and 'Content-Length' not in self.headers:
+            self.headers['Transfer-Encoding'] = 'chunked'
+
+
+class AWSResponse(object):
+
+    def __init__(self, url=None, status_code=None, headers=None, raw=None):
```

Comment:
Why are all of these defaulted to `None`?  Shouldn't these be required?

Reply:
Yeah, perhaps these just should be positional arguments.